### ISSUE TITLE
Allow sending GA events for server-side errors

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#4003]
This is related to #3896, but a bit more of a subset/prerequisite: Tracking GA events for server-side errors is not only about fatal errors. There can also be failing Google API requests that are made from the server, decoupled from a client-side API request that could handle it, see for example https://github.com/google/site-kit-wp/pull/3979#discussion_r704651187.

We should implement the PHP infrastructure needed to send GA events.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* There should be PHP infrastructure to send a GA tracking event, probably best in a `Tracking::send_event` method.
    * The method should only send the event if the current user has opted into tracking. Otherwise it should just silently bail.
    * Let's figure out how this can be accomplished. @eugene-manuilov dug up some [blog post about what looks like a hack](https://metamug.com/article/productivity/php-send-google-analytics-event.html) in https://github.com/google/site-kit-wp/issues/3896#issuecomment-901238546, and that kind of looks okay but maybe there are also some more official instructions we can follow.
* An `api_error` category event should be sent, with data in a similar format like it would be for those events on the client-side, when the Idea Hub `platforms_properties_ideaActivities::create` call fails. It should use the event name `POST:modules/idea-hub/data/create-idea-activity` (even though that isn't technically a datapoint we have).
* An event with name `fetch_token_error` and category `php_error` should be sent whenever fetching a new token fails (either in the OAuth redirect flow or when refreshing), probably best in the `OAuth_Client_Base::handle_fetch_token_exception` method.

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by felixarntz** [comment]
Any additional thoughts on this @eugene-manuilov? Let's dig around a bit more to find better instructions on how to send server-side GA events.

**Comment by eugene-manuilov** [comment]
@felixarntz if we switch to use GA4 for events tracking, then I think we can leverage the [Measurement Protocol for GA 4](https://developers.google.com/analytics/devguides/collection/protocol/ga4) that allows developers to make HTTP requests to send server-side events directly to Google Analytics servers. See the [documentation](https://developers.google.com/analytics/devguides/collection/protocol/ga4/reference), what do you think?

**Comment by eugene-manuilov** [comment]
Yeah, this is a good point, @aaemnnosttv. In this case, we will need to implement a new endpoint on the service side.
